sdks/python/apache_beam/transforms/core.py

```diff
@@ -1118,7 +1118,7 @@ def _clone(self, **extra_attributes):
     return clone
 
   def with_fanout(self, fanout):
-    return self._clone(fanout=fanout)
+    return self._clone(fanout=self.fanout)
```

Comment:
We don't want "self." in front of the fanout here (since it is now being ignored).

The test still pass of course, but the "optimization" (ie the fanout itself) is lost.

My other comment was referring to a separate "self.fanout" issue, which has however been fixed in https://github.com/apache/beam/pull//5800.

Could we fix the issue here?

/cc @aaltay